### ISSUE TITLE
3rd Party Problem

### ISSUE THREAD

**Issue by RheineAdithia** [jitsi/jitsi-meet#3010]
I've a situation where if a 3rd party enter, the screen goes to the avatar icon and all video call are dropped. The mic dropped too after the 3rd party enter. but it seems that the chat is working fine.

I've read a couple of people has the same issue just like mine. and the jitsi i will be using is running behind NAT (443, 4443, and 10000udp are already opened and redirected to the public IP address). but it still doesn't work after all the experiment i've tried from all threads.. help

**Comment by saghul** [comment]
If you are behind NAT, have you enabled the proper ice4j harvester in the videobridge config? https://github.com/jitsi/ice4j/blob/master/doc/configuration.md#mapping-harvesters

**Comment by RheineAdithia** [comment]
@saghul yes i already added the 2 line that need to be added like in the quick installation page, those 2 lines are : 
org.ice4j.ice.harvest.NAT_HARVESTER_LOCAL_ADDRESS=192.168.123.55
org.ice4j.ice.harvest.NAT_HARVESTER_PUBLIC_ADDRESS=203.127.xxx.xxx

i've installed it in different server and it works well before, but after the changes the conference doesn't work, only the p2p connection works.

and i've read from other threads that what is making this happen is the selfsigned certificate, but i didn't do that part and the problem still occur

**Comment by saghul** [comment]
What errors do you see in the JS console? Also, please show us your jicofo and jvb logs.

**Comment by RheineAdithia** [comment]
where can i see those thing mr @saghul

**Comment by RheineAdithia** [comment]
here you go @saghul 

[zandenapps.com-1527041181020.zip](https://github.com/jitsi/jitsi-meet/files/2029241/zandenapps.com-1527041181020.zip)

**Comment by RheineAdithia** [comment]
given updates on this matter, actually if 2 person join the calls using different network, the call won't commence. it only give us a blank screen. The chat feature works fine, only the video is not working

**Comment by saghul** [comment]
I see a jicofo SSL cert error, did you have this installation and have upgraded recently?

Try running this: `sudo update-ca-certificates`

Then restart jicofo.

**Comment by RheineAdithia** [comment]
@saghul i just follow the quick installation from the jitsi docs.
and running those command you gave me didn't give me any lucks on calling my partner.
any more suggestion mr @saghul?

**Comment by airhead-dom** [comment]
@RheineAdithia @saghul im having exactly the same issue. this is kinda frustates me. been trying to rresolve the issue for the past 2 weeks and still not finding a clue. heeelp mr @saghul

**Comment by damencho** [comment]
The command is update-ca-certificates -f, ddoes that change anything?

**Comment by damencho** [comment]
@airhead-dom Do you see the same in jicofo logs?

@RheineAdithia here is something to check:
https://github.com/jitsi/jitsi-meet/issues/2851#issuecomment-384454403
I'm also interested what do you have in those configs, what is missing, how did you install, did you remove anything and re-install, basically the steps to reproduce the problem so we can fix it. Recently we pushed some changes which fix most of the cases we saw.

And there is a workaround add the following config to /etc/jits/jicofo/sip-communicator.properties
org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED=true

**Comment by RheineAdithia** [comment]
@damencho first thing first, thanks for the reply mr damencho.

running update-ca-certificates -f doesn't do any luck too mr damencho, peer to peer connection still couldn't be established either. 
And for the step how i installed it, i have 2 running linux server that i use to develop with jitsi. And for the one that i give you the log, the server is a fresh server, just installed ubuntu, nginx and then run all the step in the quick installation guides. i only followed the step before the "Adding sip-gateway to Jitsi Meet" segment. But after those step, i cannot call with three person on the room. After that i tried to reinstall it by uninstalling nginx and unistalling jitsi, but still no connection can be established. And after that i tried to remove the /etc/jitsi and /etc/nginx folder too recreate it when installing nginx and jitsi, but still no luck. and for the certificate, i tried using the "/usr/share/jitsi-meet/scripts/install-letsencrypt-cert.sh", no certificate whatsoever and trying with a free certificate, and still no luck too on running jitsi on my server. 

with this comment, i attached the latest log from the JS, it showed 2 error saying these line : 
```
Logger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
```

hope this help to perfect jitsi, it is such a good apps, thats why im not stopping on experimenting ways to work things out.
[log-js.txt](https://github.com/jitsi/jitsi-meet/files/2040124/log-js.txt)

**Comment by RheineAdithia** [comment]
new error came out after i add the : 
org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED=true
[log-js-2.txt](https://github.com/jitsi/jitsi-meet/files/2040201/log-js-2.txt)

**Comment by damencho** [comment]
You have: CONFERENCE FAILED: conference.videobridgeNotAvailable
Is video bridge running?

**Comment by RheineAdithia** [comment]
yes it is running @damencho, and after i remove the org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED=true from jicofo sip-comunicator.properties, the console error disappear

**Comment by RheineAdithia** [comment]
ok after meddling with the certificate, i use the one that i get online, and p2p connection can be established, but again the problem resurface, if 3 person are in a room it raised the same error, this is it : 
it kept giving play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD is it because the ssl that you mentioned above?


Logger.js

```
> :125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> n @ Logger.js:125
> AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> Logger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> n @ Logger.js:125
> AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> Logger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> n @ Logger.js:125
> AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> 2Logger.js:125 [modules/statistics/AnalyticsAdapter.js] <e.value>:  Not sending an event, disposed.
> n @ Logger.js:125
> Logger.js:125 [modules/xmpp/ChatRoom.js] <t.value>:  entered test@conference.zandenapps.com/5d441f04 Object
> Logger.js:125 [modules/UI/videolayout/VideoLayout.js] <Object.changeUserAvatar>:  Missed avatar update - no small video yet for 5d441f04
> n @ Logger.js:125
> Logger.js:125 [conference.js] <n.<anonymous>>:  USER %s connnected 5d441f04 e
> Logger.js:125 [JitsiConference.js] <r._maybeStartOrStopP2P>:  Will stop P2P with: test@conference.zandenapps.com/fc8a4e23
> Logger.js:125 [JitsiConference.js] <r._removeRemoteTracks>:  Removing remote P2P track: RemoteTrack[fc8a4e23, audio, p2p: true]
> Logger.js:125 [modules/UI/videolayout/RemoteVideo.js] <r.removeRemoteStreamElement>:  Audio removed fc8a4e23 ge.fn.init(1)
> Logger.js:125 [JitsiConference.js] <r._removeRemoteTracks>:  Removing remote P2P track: RemoteTrack[fc8a4e23, video, p2p: true]
> Logger.js:125 [modules/UI/videolayout/RemoteVideo.js] <r.removeRemoteStreamElement>:  Video removed fc8a4e23 ge.fn.init(1)
> Logger.js:125 [JitsiConference.js] <r._stopP2PSession>:  Stopping remote stats for P2P connection
> Logger.js:125 [JitsiConference.js] <r._stopP2PSession>:  Stopping CallStats for P2P connection
> Logger.js:125 [modules/xmpp/JingleSessionPC.js] <t.value>:  Sending session-terminate <iq to=​"test@conference.zandenapps.com/​fc8a4e23" type=​"set" xmlns=​"jabber:​client" id=​"e0abadc1-90f1-47e7-8fcb-6a8ef716fd67:​sendIQ">​…​</iq>​
> Logger.js:125 [modules/xmpp/JingleSessionPC.js] <t.value>:  Session terminated JingleSessionPC[p2p=true,initiator=false,sid=7e1c9661c3fc] undefined undefined
> Logger.js:125 [modules/RTC/TraceablePeerConnection.js] <r.close>:  Closing TPC[1,p2p:true]...
> Logger.js:125 [JitsiConference.js] <r._setP2PStatus>:  Peer to peer connection closed!
> Logger.js:125 [JitsiConference.js] <r._stopP2PSession>:  Not adding remote JVB tracks - no session yet
> Logger.js:125 [modules/UI/videolayout/LargeVideoManager.js] <>:  hover in %s fc8a4e23
> Logger.js:125 [modules/xmpp/JingleSessionPC.js] <t.value>:  The session has ended - cancelling action: oniceconnectionstatechange
> Logger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> n @ Logger.js:125
> AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> Logger.js:125 [modules/xmpp/strophe.jingle.js] <a.value>:  on jingle session-terminate from test@conference.zandenapps.com/fc8a4e23 <iq xmlns=​"jabber:​client" type=​"set" to=​"64f58c84-ddb0-42dc-80c0-2139fc8e7d1b@zandenapps.com/​9a560083-cab9-4dec-a316-c372b3879732" from=​"test@conference.zandenapps.com/​fc8a4e23" id=​"NjRmNThjODQtZGRiMC00MmRjLTgwYzAtMjEzOWZjOGU3ZDFiQHphbmRlbmFwcHMuY29tLzlhNTYwMDgzLWNhYjktNGRlYy1hMzE2LWMzNzJiMzg3OTczMgAyMDMzYjQyZi0zZmU2LTQ0MGYtYTMxNC1hMTNlZDY1MGQxMWU6c2VuZElRAEZBBehSvXjoKJm68uNAYAE=">​…​</iq>​
> Logger.js:125 [modules/xmpp/strophe.jingle.js] <a.value>:  invalid session id <iq xmlns=​"jabber:​client" type=​"set" to=​"64f58c84-ddb0-42dc-80c0-2139fc8e7d1b@zandenapps.com/​9a560083-cab9-4dec-a316-c372b3879732" from=​"test@conference.zandenapps.com/​fc8a4e23" id=​"NjRmNThjODQtZGRiMC00MmRjLTgwYzAtMjEzOWZjOGU3ZDFiQHphbmRlbmFwcHMuY29tLzlhNTYwMDgzLWNhYjktNGRlYy1hMzE2LWMzNzJiMzg3OTczMgAyMDMzYjQyZi0zZmU2LTQ0MGYtYTMxNC1hMTNlZDY1MGQxMWU6c2VuZElRAEZBBehSvXjoKJm68uNAYAE=">​…​</iq>​
> n @ Logger.js:125
> 3Logger.js:125 [modules/statistics/AnalyticsAdapter.js] <e.value>:  Not sending an event, disposed.
> n @ Logger.js:125
> Logger.js:125 [modules/UI/videolayout/LargeVideoManager.js] <>:  hover in %s 5d441f04
> Logger.js:125 [modules/statistics/AnalyticsAdapter.js] <e.value>:  Not sending an event, disposed.
> n @ Logger.js:125
> Logger.js:125 [modules/UI/videolayout/LargeVideoManager.js] <>:  hover in %s fc8a4e23
> logger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDn @ Logger.js:125AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDLogger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDn @ Logger.js:125AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDLogger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDn @ Logger.js:125AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD2Logger.js:125 [modules/statistics/AnalyticsAdapter.js] <e.value>:  Not sending an event, disposed.n @ Logger.js:125Logger.js:125 [modules/xmpp/ChatRoom.js] <t.value>:  entered test@conference.zandenapps.com/5d441f04 ObjectLogger.js:125 [modules/UI/videolayout/VideoLayout.js] <Object.changeUserAvatar>:  Missed avatar update - no small video yet for 5d441f04n @ Logger.js:125Logger.js:125 [conference.js] <n.<anonymous>>:  USER %s connnected 5d441f04 eLogger.js:125 [JitsiConference.js] <r._maybeStartOrStopP2P>:  Will stop P2P with: test@conference.zandenapps.com/fc8a4e23Logger.js:125 [JitsiConference.js] <r._removeRemoteTracks>:  Removing remote P2P track: RemoteTrack[fc8a4e23, audio, p2p: true]Logger.js:125 [modules/UI/videolayout/RemoteVideo.js] <r.removeRemoteStreamElement>:  Audio removed fc8a4e23 ge.fn.init(1)Logger.js:125 [JitsiConference.js] <r._removeRemoteTracks>:  Removing remote P2P track: RemoteTrack[fc8a4e23, video, p2p: true]Logger.js:125 [modules/UI/videolayout/RemoteVideo.js] <r.removeRemoteStreamElement>:  Video removed fc8a4e23 ge.fn.init(1)Logger.js:125 [JitsiConference.js] <r._stopP2PSession>:  Stopping remote stats for P2P connectionLogger.js:125 [JitsiConference.js] <r._stopP2PSession>:  Stopping CallStats for P2P connectionLogger.js:125 [modules/xmpp/JingleSessionPC.js] <t.value>:  Sending session-terminate <iq to=%E2%80%8B"test@conference.zandenapps.com/%E2%80%8Bfc8a4e23" type=%E2%80%8B"set" xmlns=%E2%80%8B"jabber:%E2%80%8Bclient" id=%E2%80%8B"e0abadc1-90f1-47e7-8fcb-6a8ef716fd67:%E2%80%8BsendIQ">%E2%80%8B%E2%80%A6%E2%80%8B</iq>%E2%80%8BLogger.js:125 [modules/xmpp/JingleSessionPC.js] <t.value>:  Session terminated JingleSessionPC[p2p=true,initiator=false,sid=7e1c9661c3fc] undefined undefinedLogger.js:125 [modules/RTC/TraceablePeerConnection.js] <r.close>:  Closing TPC[1,p2p:true]...Logger.js:125 [JitsiConference.js] <r._setP2PStatus>:  Peer to peer connection closed!Logger.js:125 [JitsiConference.js] <r._stopP2PSession>:  Not adding remote JVB tracks - no session yetLogger.js:125 [modules/UI/videolayout/LargeVideoManager.js] <>:  hover in %s fc8a4e23Logger.js:125 [modules/xmpp/JingleSessionPC.js] <t.value>:  The session has ended - cancelling action: oniceconnectionstatechangeLogger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDn @ Logger.js:125AbstractAudio.js:83 Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDDLogger.js:125 [modules/xmpp/strophe.jingle.js] <a.value>:  on jingle session-terminate from test@conference.zandenapps.com/fc8a4e23 <iq xmlns=%E2%80%8B"jabber:%E2%80%8Bclient" type=%E2%80%8B"set" to=%E2%80%8B"64f58c84-ddb0-42dc-80c0-2139fc8e7d1b@zandenapps.com/%E2%80%8B9a560083-cab9-4dec-a316-c372b3879732" from=%E2%80%8B"test@conference.zandenapps.com/%E2%80%8Bfc8a4e23" id=%E2%80%8B"NjRmNThjODQtZGRiMC00MmRjLTgwYzAtMjEzOWZjOGU3ZDFiQHphbmRlbmFwcHMuY29tLzlhNTYwMDgzLWNhYjktNGRlYy1hMzE2LWMzNzJiMzg3OTczMgAyMDMzYjQyZi0zZmU2LTQ0MGYtYTMxNC1hMTNlZDY1MGQxMWU6c2VuZElRAEZBBehSvXjoKJm68uNAYAE=">%E2%80%8B%E2%80%A6%E2%80%8B</iq>%E2%80%8BLogger.js:125 [modules/xmpp/strophe.jingle.js] <a.value>:  invalid session id <iq xmlns=%E2%80%8B"jabber:%E2%80%8Bclient" type=%E2%80%8B"set" to=%E2%80%8B"64f58c84-ddb0-42dc-80c0-2139fc8e7d1b@zandenapps.com/%E2%80%8B9a560083-cab9-4dec-a316-c372b3879732" from=%E2%80%8B"test@conference.zandenapps.com/%E2%80%8Bfc8a4e23" id=%E2%80%8B"NjRmNThjODQtZGRiMC00MmRjLTgwYzAtMjEzOWZjOGU3ZDFiQHphbmRlbmFwcHMuY29tLzlhNTYwMDgzLWNhYjktNGRlYy1hMzE2LWMzNzJiMzg3OTczMgAyMDMzYjQyZi0zZmU2LTQ0MGYtYTMxNC1hMTNlZDY1MGQxMWU6c2VuZElRAEZBBehSvXjoKJm68uNAYAE=">%E2%80%8B%E2%80%A6%E2%80%8B</iq>%E2%80%8Bn @ Logger.js:1253Logger.js:125 [modules/statistics/AnalyticsAdapter.js] <e.value>:  Not sending an event, disposed.n @ Logger.js:125Logger.js:125 [modules/UI/videolayout/LargeVideoManager.js] <>:  hover in %s 5d441f04Logger.js:125 [modules/statistics/AnalyticsAdapter.js] <e.value>:  Not sending an event, disposed.n @ Logger.js:125Logger.js:125 [modules/UI/videolayout/LargeVideoManager.js] <>:  hover in %s fc8a4e23
```

**Comment by damencho** [comment]
org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED=true  has nothing to do with jvb not registered to xmpp server.
 In 3 person conference if having problem you need to check jicofo logs, if you see the cert error you need to fix the certificates, to make auth.your.domain cert trusted on the server running jicofo or use org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED=true to disable cert checking.

**Comment by c526537207** [comment]
This problem I have encountered is a problem when p2p switches to JVB.Is your service built on the server above?

**Comment by scrolls-mckenzie** [comment]
Did this ever get solved?

I'm still having issues with this. I have a brand new EC2 in a VPC and have forwarded the ports in the security group as follows

<img width="932" alt="security-group" src="https://user-images.githubusercontent.com/14047475/44294396-cadec480-a2d9-11e8-8924-dd40e2c59551.png">

I have also added the private an public IP's of my instance into the sip.communicator-properties
`org.ice4j.ice.harvest.NAT_HARVESTER_LOCAL_ADDRESS=172.0.0.xxx` `org.ice4j.ice.harvest.NAT_HARVESTER_PUBLIC_ADDRESS=13.xxx.xxx.xxx`

I can have 2 participants using P2P without issue but when a third joins I get no connections for audio or video. In the config file, if I turn P2P off I cannot get 2 participants to work which leads me to believe it is the JVB. I have tried browser to browser as well as using an electron package as a client.

The issue that i see when i connect the third participant in the Chrome Console is

`[modules/xmpp/strophe.jingle.js] <a.value>: invalid session id ​`

I have read dozens of similar posts that lead to people getting it to work due to ports but I think i have the covered. Any comments would be really appreciated

**Comment by saghul** [comment]
Please paste full console log.

**Comment by maheshzh** [comment]
Hi,

Is this problem reported in k8s env?. Recently we configured jitsimeet on kubernates cluster. Nginx access is used to connect to meet and we opened udp port through node port. Everything works fine except mobile. If I log in on mobile a third party (Fellow Jitster, without audio and video) automatically logs in and leaves the meeting. We checked the jicofo logs but it looks like a normal user

Pleae help me
